point.go

```diff
@@ -91,6 +94,38 @@ func (g *Point) Swap(g2 *Point) {
 	*g, *g2 = *g2, *g
 }
 
+func (g *Point) Buffer(numOfPoints int, distance float64) (*Polygon, error) {
+	if g.layout != XY {
+		return nil, errors.New("currently only supports 2d points")
+	}
+
+	if numOfPoints < 3 {
+		return nil, errors.New("numOfPoint must be more than 3")
+
+	}
+
+	coords := []Coord{}
+	angleIncrement := (float64(2) * math.Pi) / float64(numOfPoints)
+	var angle float64
+	var coordToAdd Coord
+	for i := 0; i < numOfPoints; i++ {
+		angle = float64(i) * angleIncrement
```

Comment:
This will lose precision as any error in `angleIncrement` will be accumulated `numOfPoints` times. It's better to write this as

```go
angle = 2 * math.Pi * float(i) / float(numOfPoints)
```

which will avoids accumulating errors.

Reply:
hey, sure! in terms of coding styles i will fix.
about the precision error i have no problem changing it but the min tolerated error in the test stays the same